framework/client.py

```diff
@@ -96,18 +98,34 @@ def __on_finish(self):
         if not hasattr(self.proc, "terminate"):
             return
         tf_cfg.dbg(3, "Stopping client")
+
+        t0 = time.time()
+        while True:
+            try:
+                t = time.time()
+                if t - t0 > self.duration:
+                    break
+                self.proc_results = self.resq.get_nowait()
+                break
+            except queue.Empty:
+                time.sleep(0.01)
+
         self.proc.terminate()
-        if not self.resq.empty():
-            self.proc_results = self.resq.get()
         self.proc = None
 
-        if self.proc_results != None:
+        if self.proc_results:
             tf_cfg.dbg(3, '\tclient stdout:\n%s' % self.proc_results[0].decode())
 
             if len(self.proc_results[1]) > 0:
                 tf_cfg.dbg(2, '\tclient stderr:\n%s' % self.proc_results[1].decode())
 
             self.parse_out(self.proc_results[0], self.proc_results[1])
+        else:
+            tf_cfg.dbg(
+                2,
+                f'\tCmd command "{self.form_command()}" has not received data from queue. '
```

Comment:
Perhaps better to use `{self.cmd}`? `form_command()` could have side effects, and for `Wrk` second call will double the options.